src/http-requests.ts

```diff
@@ -20,16 +21,34 @@ function constructHostURL(host: string): string {
   }
 }
 
+function createHeaders(config: Config): Record<string, any> {
+  config.headers = Object.assign({}, config.headers || {}) // assign to avoid referencing)
+  const defaultHeaders = {
+    'X-Meilisearch-Client': `Meilisearch JS (v${PACKAGE_VERSION})`,
+    'Content-Type': 'application/json',
+  }
+  const headers: Record<string, any> = {}
+
+  if (config.apiKey) {
+    headers['Authorization'] = `Bearer ${config.apiKey}`
+  }
+
+  if (config.headers['X-Meilisearch-Client']) {
```

Comment:
I think this should be an array, something like `config.clientHeaders`, and inside of the client, we handle the `.headers`.

To improve my explanation:

`Config` object has a `clientHeaders` which is supposed to be an array.

After that, we will get the `clientHeaders` and add our default headers to it, which will be `Meilisearch JavaScript (v${PACKAGE_VERSION})`.

So you will be able to have a list and with this list join every header with `;`.

Why does it like this?

Because if we handle it as an array we will be able to support multiple overloaded clients not just two.

`X-Meilisearch-Client: "Meilisearch JavaScript (v0.0.1); Meilisearch Instant-Meilisearch (v0.0.1); Meilisearch Strapi (v0.0.1)"`.

Reply:
Is this object in the current public API of this package? 
```js
  headers: {
    something: ""
  }
```

Because yes, I would like to have a key/value in the root level! 

```js
client = new Meilisearch({
  host: "...", 
  clientHeaders: ["my lib vx", "my other lib vx.x.x"]
})
```

Why I don't like the `"X-Meilisearch-Client"` exposed? because if in the future the engine decides to change this `"X-Meilisearch-Client"` to another name, for some reason we are bound with this name, and in this case, we will have to change all the way down. It is just to hide this _internal_ name.